src/RecoveryServices/RecoveryServices/Vault/GetAzureRMRecoveryServicesVaultSettingsFile.cs

```diff
@@ -559,19 +557,31 @@ private string GenerateVaultCreds(X509Certificate2 cert, string subscriptionId,
         /// <summary>
         /// Generates vault creds file content for backup Vault
         /// </summary>
-        /// <param name="certificateString">management certificate</param>
         /// <param name="subscriptionId">subscription Id</param>
         /// <param name="vaultCertificateResponse"></param>
         /// <returns>xml file in string format</returns>
-        private string GenerateVaultCredsForBackup(string certificateString, string subscriptionId,
+        private string GenerateVaultCredsForBackup(string subscriptionId,
             VaultCertificateResponse vaultCertificateResponse)
         {
+            string certificateString = Convert.ToBase64String(
+                vaultCertificateResponse.Properties.Certificate);
+
             using (var output = new MemoryStream())
             {
                 using (var writer = XmlWriter.Create(output, GetXmlWriterSettings()))
                 {
                     var aadDetails = vaultCertificateResponse.Properties as ResourceCertificateAndAadDetails;
 
+                    string aadAudience = aadDetails.AadAudience;
+                    if (string.IsNullOrEmpty(aadAudience))
+                    {
+                        aadAudience = string.Format(CultureInfo.InvariantCulture,
```

Comment:
This should be extracted as a method to avoid duplication with L654-L658 for example
```suggestion
                        aadAudience = GetAadAudience(Vault.Location, ......)
```